draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,621 @@
+
+;; ADDITIONAL SECTION:
+ns.customer5.example.   3600    IN      A       198.51.100.5
+ns.customer5.example.   3600    IN      AAAA    2001:db8:5::1
+
+;; Query time: 0 msec
+;; EDNS: version 0; flags: do ; udp: 1232
+;; SERVER: 192.0.2.53
+;; WHEN: Mon Jul  1 20:36:25 2024
+;; MSG SIZE  rcvd: 456
+~~~
+{: #deleg-response title="An incremental deleg referral response"}
+
+The referral response in {{deleg-response}} includes the signed SVCB RRset in the authority section.
+
+As another example, querying the zone from {{dnssec-zone}} for `www.customer6.example. A`, will return the following referral response:
+
+~~~
+;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 36574
+;; flags: qr ; QUERY: 1, ANSWER: 0, AUTHORITY: 9, ADDITIONAL: 2
+;; QUESTION SECTION:
+;; www.customer6.example.       IN      A
+
+;; ANSWER SECTION:
+
+;; AUTHORITY SECTION:
+customer6.example.      3600    IN      NS      ns.customer6.example.
+customer6.example.      3600    IN      DS      ...
+customer6.example.      3600    IN      RRSIG   DS ...
+customer5._deleg.example.       1234    IN      NSEC    (
+                customer7._deleg.example.  RRSIG NSEC SVCB )
+customer5._deleg.example.       1234    IN      RRSIG   NSEC ...
+example.        1234    IN      NSEC    (
+                customer5._deleg.example.  NS SOA RRSIG NSEC DNSKEY )
+example.        1234    IN      RRSIG   NSEC ...
+example.        1234    IN      SOA     ns.example. (
+                zonemaster.example. 20240627 7200 3600 1209600 1234 )
+example.        1234    IN      RRSIG   SOA ...
+
+;; ADDITIONAL SECTION:
+ns.customer6.example.   3600    IN      A       203.0.113.1
+ns.customer6.example.   3600    IN      AAAA    2001:db8:6::1
+
+;; Query time: 0 msec
+;; EDNS: version 0; flags: do ; udp: 1232
+;; SERVER: 192.0.2.53
+;; WHEN: Tue Jul  2 10:23:53 2024
+;; MSG SIZE  rcvd: 744
+~~~
+{: #no-incr-deleg-response title="Referral response without incremental deleg"}
+
+Next to the legacy delegation, the incremental deleg supporting authoritative returns the NSEC(3) RRs needed to show that there was no incremental delegation in the referral response in {{no-incr-deleg-response}}.
+
+Querying the zone from {{dnssec-zone}} for `www.customer7.example. A`, will return the following referral response:
+
+~~~
+;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 9456
+;; flags: qr ; QUERY: 1, ANSWER: 0, AUTHORITY: 7, ADDITIONAL: 2
+;; QUESTION SECTION:
+;; www.customer7.example.       IN      A
+
+;; ANSWER SECTION:
+
+;; AUTHORITY SECTION:
+customer7.example.      3600    IN      NS      ns.customer5.example.
+customer7.example.      3600    IN      DS      ...
+customer7.example.      3600    IN      RRSIG   DS ...
+customer7._deleg.example.       3600    IN      CNAME   (
+                customer5._deleg.example. )
+customer7._deleg.example.       3600    IN      RRSIG   CNAME ...
+customer5._deleg.example.       3600    IN      SVCB    1 (
```

Comment:
How does this work. Does the target of the CNAME have to have a _dns prefix?

Reply:
Not in this case, because in this case it is referring to another _deleg set.